src/System.Management.Automation/engine/pipeline.cs

```diff
@@ -220,44 +220,33 @@ private void Log(string logElement, InvocationInfo invocation, PipelineExecution
                 }
             }
 
-            if (!string.IsNullOrEmpty(logElement))
+            if (_needToLog && !string.IsNullOrEmpty(logElement))
             {
+                _eventLogBuffer ??= new List<string>();
                 _eventLogBuffer.Add(logElement);
```

Comment:
I still suggest to think about intermediate flushing the buffer if _eventLogBuffer.Count > Threshold. This will keep the buffer size small for long running pipelines that original issue is about.

Reply:
As is called out in the PR description, this is "_a partial fix for #15341_".
The fundamental flaw I pointed out in https://github.com/PowerShell/PowerShell/issues/15341#issuecomment-831549326 will be addressed in a separate PR -- it will require 1) defining the right threshold, 2) make sure events with partial payload are correlated.